### ISSUE TITLE
Incompatible with PSR/Cache ^2

### ISSUE THREAD

**Issue by mangelsnc** [WhichBrowser/Parser-PHP#659]
Hi!

If you try to install this package in a Symfony 5.x environment with Doctrine 2.9, you get the next error:

```
Problem 1
    - Root composer.json requires whichbrowser/parser v2.1.2 -> satisfiable by whichbrowser/parser[v2.1.2].
    - whichbrowser/parser v2.1.2 requires psr/cache ^1.0 -> found psr/cache[1.0.0, 1.0.1] but the package is fixed to 2.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
```

I think it can be solved by downgrading the version of `psr/cache`, but I think it would be better to make this library compatible with it.

This is the exit of `composer why psr/cache`:

```
$ composer why psr/cache
doctrine/annotations     1.13.1   requires   psr/cache (^1 || ^2 || ^3)
doctrine/cache           2.0.3    conflicts  psr/cache (>=3)
doctrine/orm             2.9.2    requires   psr/cache (^1 || ^2 || ^3)
doctrine/persistence     2.2.1    requires   psr/cache (^1.0|^2.0|^3.0)
symfony/cache            v5.2.10  requires   psr/cache (^1.0|^2.0)
symfony/cache-contracts  v2.4.0   requires   psr/cache (^1.0|^2.0|^3.0)
```

It seems like all the dependencies of `psr/cache` will work with version `^1`, but I think it would be better to upgrade the dependency of this library.

**Comment by therealsamsquanch** [comment]
Pull request #665 adds compatibility with psr/cache v2 and v3. Unfortunately, it is still waiting for approval. 

@NielsLeenheer can you please check the pull-requests awaiting approval?

**Comment by rogeriolino** [comment]
for while, to solve it, you can use the fork branch in your project:

```json
{
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/javer/Parser-PHP"
        }
    ],
    "require": {
        "whichbrowser/parser": "dev-psr-cache"
    }
}
```

**Comment by summercms** [comment]
@therealsamsquanch 

`psr/cache 2.0 and 3.0` require `php: >=8.0.0`

See here as proof: https://packagist.org/packages/psr/cache

This repo says: https://github.com/WhichBrowser/Parser-PHP/blob/bcf642a1891032de16a5ab976fd352753dd7f9a0/composer.json#L17

p.s. I'm half way through coding this repo to php `8.0.3`

**Comment by summercms** [comment]
Working php 8.0.3 version changes needed:

```json
    "require": {
        "php": "^8.0",
        "psr/cache": "^1.0 || ^2.0 || ^3.0"
    },
    "require-dev": {
        "symfony/yaml": "^5.3",
        "phpunit/php-code-coverage": "^9.2.7",
        "phpunit/phpunit": "^8.0 || 9.*",
        "php-coveralls/php-coveralls": "^2.4.3",
        "enso-media/reverse-regex": "^0.2.0",
        "squizlabs/php_codesniffer": "^3.6",
        "cache/array-adapter": "^1.2"
    },
```

Proof:

![image](https://user-images.githubusercontent.com/57409060/149836466-23002daf-8a71-4f75-a588-f215844aeda5.png)

**Comment by NielsLeenheer** [comment]
Thanks. Included in v2.1.3.
